mesh/nurbs.hpp

```diff
@@ -403,6 +405,21 @@ class NURBSExtension
    void UniformRefinement();
    void KnotInsert(Array<KnotVector *> &kv);
    void KnotInsert(Array<Vector *> &kv);
+
+   int GetElementPatch(int elem) const { return el_to_patch[elem]; }
+   void GetElementIJK(int elem, Array<int> & ijk);
+
+   const Array3D<int>& GetPatchDofs(const int patch);
+
+   int NumPatchDofs1D(const int patch, const int dim);
+
+private:
+   std::vector<Array3D<int>> patchDofs;
+   Array2D<int> ndof1D;
+   std::vector<std::vector<std::set<int>>> patch_ijk;
```

Comment:
Good observation, that `patch_ijk` is never used. It can be removed. I think I initially planned to use it but did not need it, and thought it may be useful in the future. I will document it, and reviewers can give opinions on whether to keep it. As for `Array2D`, the type must be "trivial", e.g. `int`, `double`, etc. There is a class `IntegerSet` in MFEM, but I have not seen it in use and am not sure whether it is as efficient as `std::set`. Maybe someone else can comment on which is preferable.